src/Type/Constant/ConstantArrayType.php

```diff
@@ -444,15 +445,32 @@ public function equals(Type $type): bool
 
 	public function isCallable(): TrinaryLogic
 	{
-		$typeAndMethods = $this->findTypeAndMethodNames();
-		if ($typeAndMethods === []) {
+		$typeAndMethodNames = $this->findTypeAndMethodNames();
+		if ($typeAndMethodNames === []) {
 			return TrinaryLogic::createNo();
 		}
 
-		$results = array_map(
-			static fn (ConstantArrayTypeAndMethod $typeAndMethod): TrinaryLogic => $typeAndMethod->getCertainty(),
-			$typeAndMethods,
-		);
+		$results = [];
+		foreach ($typeAndMethodNames as $typeAndMethodName) {
+			$results[] = $typeAndMethodName->getCertainty();
+
+			if ($typeAndMethodName->isUnknown()) {
+				continue;
+			}
+
+			$type = $typeAndMethodName->getType();
+			$methodName = $typeAndMethodName->getMethod();
+
+			if (!$type->hasMethod($methodName)->yes()) {
+				continue;
+			}
+
+			$method = $type->getMethod($methodName, new OutOfClassScope());
+			if ($method->isStatic()) {
```

Comment:
actually this should depend on php-version.
within the type-classes we don't have access to `PhpVersion` though